src/components/common/RestEntity/README.md

```diff
@@ -0,0 +1,882 @@
+
+Currently, this component worries about:
+
+1. Fetching Data — which comes with all complications discussed above
+
+2. How to present that data to our user
+
+But what if it had to worry about just one job:
+
+1. **How to present that data to our user**
+
+- ~~Fetching Data — which comes with all complications discussed above~~
+
+## The Problem
+
+In an application like Reach LMS, many of our components are going to have to worry about data fetching. There's most likely going to be at LEAST two different components for each "Category of Data," so to speak.
+
+In Reach LMS, the four biggest root types of data are:
+
+- User
+
+- Program
+
+- Course
+
+- Module
+
+Each one of those will probably be presented either as a one-off view or a list of multiple. Say that a user click on a Profile Page — now my component would have to fetch data for a **SINGLE USER**. And then that user clicks on their `ManageUsers` dashboard to view all of the users in their system; now `ManageUsers` would have to fetch data for a **LIST OF USERS**.
+
+You can probably pretty easily imagine a situation for a Single view vs a List view for each of the four big data types above.
+
+- User:
+  - Profile Page — Single User
+  - User Management Dash — List of Users
+- Program
+  - Edit Program Form — Fetch Single Program
+  - Program Dashboard — List of Programs
+- And so on, and so forth
+
+
+Handling all of that data fetching IN each component on top of managing application state, user-flow, interactivity, and everything else would result in some wildly complex components.
+
+But anything fetching data is going to do it in relatively the same way. And each component really just needs to be able to decide how to DISPLAY that information to the user. 
+
+Enter `RestEntity`. This component is all about data-fetching.
+
+---
+
+## RestEntity Overview
+
+The `RestEntity` component is a compound component that utilizes the React Context API to handle data-fetching and consume relational links. Though using such a component will feel new and different to many of you, I promise it appears more intimidating than it really is. Once you're working with it,
+it'll feel very similar to the same old React that you're used to.
+
+The whole purpose of the `RestEntity` is to simplify the process of interacting with HAL-shaped data from our RESTful, HATEOAS-oriented backend. That is to say, every single data object has hypermedia links. Instead of worrying about complex Redux thunks and/or over-the-top routing, our `RestEntity` is going to take whatever endpoint you want to feed it and get the data you care about. Then, it'll give control back to you so you can do what you're so good at doing: building components to PRESENT information.
```

Comment:
- I would position this as the first section on top before exploring the Program component example as it is a good overview and guides the reader into why this specific tool was chosen. 
- This will also encourage the reader to want to use this and spend the time carefully paying attention to the rest of the docs.

Reply:
word, that’s actually where it was to start out lol. i couldn’t decide where to start